### DISCUSSION TITLE
Does fxx=0 means actual weather reading?

### DISCUSSION THREAD

**Discussion by yinghawl** [blaylockbk/Herbie#50] [Q&A]
Hi all,

I am fairly new to using Herbie to analyze historical weather data. Does the following code (fxx=0) query the actual weather data for 2021-01-01 12:00?

```python
H = Herbie(
    "2021-01-01 12:00",
    model="hrrr",
    product="sfc",
    fxx=0
)
```

Thanks,
Anson

**Comment by blaylockbk** [comment]
Hi @yinghawl,

In Herbie, the `fxx` variable is the forecast lead time.

`fxx=0` represents **F00**, which is the forecast with zero hours lead time, also known as the "analysis." So yes, fxx=0 will query the weather analysis valid at the time you specify.

---

If you were to set `DATE="2021-01-01 12:00"`, and `fxx=6` (meaning **F06**), then you are asking for the 6 hour forecast of the model initialized at 12:00 UTC. The data would represent data valid at 18:00 (6 hours after 12:00).

**Reply by rthedin** [reply]
Hi Brian,
Thanks for creating and maintaining Herbie. I have a related question, also using HRRR data.

Using the example you gave, I can do
`Herbie(valid_date='2021-01-01 12:00', model='hrrr', product='sfc', fxx=6)`
and Herbie will print "Found 2021-Jan-01 06:00 UTC F06 [HRRR] [product=sfc] GRIB2 file from aws and index file from aws."

From the information printed, it seems like Herbie gave me data of a 6 hour forecast initialized at 6AM, so data at 12PM. That appears to be contradict what you just said-- I was expecting to see some info on 18:00. What am I missing here?

**Reply by blaylockbk** [reply]
Herbie accepts arguments for `date` or `valid_date`. 
- `date` represents the model initialization time
- `valid_date` represents the datetime the data is valid

https://github.com/blaylockbk/Herbie/blob/b96ce71361fb9f896b9352e41514fdaeea7ca62c/herbie/archive.py#L250-L258

If you want the data valid at 18z from the model initialized at 12z you would do

```python
Herbie(date='2021-01-01 12:00', model='hrrr', product='sfc', fxx=6)
```
or 
```python
Herbie(valid_date='2021-01-01 18:00', model='hrrr', product='sfc', fxx=6)
```

**Reply by rthedin** [reply]
That makes sense. Thanks for the quick response!
